aten/src/ATen/Context.cpp

```diff
@@ -225,17 +229,8 @@ bool Context::hasMKLDNN() {
 }
 
 bool Context::hasMPS() {
-#if defined(__APPLE__)
-#if __is_target_os(macOS)
-  _Pragma("clang diagnostic ignored \"-Wunsupported-availability-guard\"")
-  if (__builtin_available(macOS 12.3, *) || __builtin_available(macOSApplicationExtension 12.3, *)) {
-    return c10::impl::hasDeviceGuardImpl(at::DeviceType::MPS);
-  } else {
-    return false;
-  }
-#else
-  return false;
-#endif
+#if USE_MPS
```

Comment:
This is not dynamic check anymore, and since we are building on 12.3+, having this macro will it make a difference ?. 
Are we planning to use `@available` here to see if we still have these issues?

Reply:
Actually, it is a dynamic check now, as `hasMPS` would return true if and only if there is a Metal device and code is running on MacOS 12.3 (which I'm using `[[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:v12_3]`  to check, as `@available` can not be used to check if OS version is lower than something)